ansible_base/rbac/permission_registry.py

```diff
@@ -22,43 +22,59 @@
 logger = logging.getLogger('ansible_base.rbac.permission_registry')
 
 
+class ModelPermissionInfo:
+    """Container of RBAC registration information for a model in permission_registry"""
+
+    def __init__(self, model, parent_field_name='organization', allow_object_roles=True):
+        self.model_name = model._meta.model_name
+        self.app_label = model._meta.app_label
+        if parent_field_name == self.model_name:
+            # model can not be its own parent
+            self.parent_field_name = None
+        else:
+            self.parent_field_name = parent_field_name
+        self.allow_object_roles = allow_object_roles
+        self.model = model
+
+
 class PermissionRegistry:
     def __init__(self):
-        self._registry = set()  # model registry
-        self._name_to_model = dict()
-        self._parent_fields = dict()
+        self._registry = dict()  # model registry
         self._managed_roles = dict()  # code-defined role definitions, managed=True
         self.apps_ready = False
         self._tracked_relationships = set()
         self._trackers = dict()
 
-    def register(self, *args, parent_field_name='organization'):
+    def register(self, *args, **kwargs):
         if self.apps_ready:
             raise RuntimeError('Cannot register model to permission_registry after apps are ready')
-        for cls in args:
-            if cls not in self._registry:
-                self._registry.add(cls)
-                model_name = cls._meta.model_name
-                if model_name in self._name_to_model:
-                    raise RuntimeError(f'Two models registered with same name {model_name}')
-                self._name_to_model[model_name] = cls
-                if model_name != 'organization':
-                    self._parent_fields[model_name] = parent_field_name
+        for model in args:
+            if model._meta.model_name not in self._registry:
+                info = ModelPermissionInfo(model, **kwargs)
+                self._registry[info.model_name] = info
+            elif self._registry[model._meta.model_name] is model:
```

Comment:
This will account for the app_name because we are comparing the whole model right?

Reply:
It will yell at you if you register 2 models with the same name, even if they're in different apps. Proxy models are fine, but don't register the original... and also register the proxy.